### PR TITLE
[ModSupport & ModManager] Edit launch logic and create shortcut button

### PR DESCRIPTION
**Pull request by shoushou1106**
Add PluginAction for ModManager `ManageModDataWindow`
Add "Launch with previous data" Button to Editor `MainWindow`
图片
Edit ModExecutor launch logic to apply this [commit](https://github.com/CadeEvs/FrostyToolsuite/commit/7f27afe0834fd193ccceaf61e178e1f95d8878ee) changes elsewhere

### CODE REVIEW

**Comment by Dyvinia** [comment]
i feel like itd be better if the button didnt have text like the launch without debugging button in VS

**Comment by shoushou1106** [comment]
> i feel like itd be better if the button didnt have text like the launch without debugging button in VS

The icon is actually small and difficult to see clearly, but it is indeed better without text.

**Comment by shoushou1106** [comment]
ok i delete the text
图片
  
and i add a button that can create shortcut to launch game with moddata
图片

**Comment by Dyvinia** [comment]
> ok i delete the text 图片
> 
> and i add a button that can create shortcut to launch game with moddata 图片

the X button should be at the end (very right) as typically thats where a list would have the button to remove items

**Comment by shoushou1106** [comment]
> the X button should be at the end (very right) as typically thats where a list would have the button to remove items

ok it look like this now
img

File: FrostyModManager/FrostyModManager.csproj [hunk 1]
```diff
@@ -301,6 +302,17 @@
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </Content>
   </ItemGroup>
+  <ItemGroup>
```
Comment (wannkunstbeikor) [inline FrostyModManager/FrostyModManager.csproj:305]: what is this for?

File: FrostyModManager/Windows/ManageModDataWindow.xaml.cs [hunk 2]
```diff
@@ -122,9 +130,117 @@ private void openModData_Click(object sender, RoutedEventArgs e)
...
-            FrostyModExecutor.ExecuteProcess($"{Path.GetDirectoryName(getModDataPath())}\\{ProfilesLibrary.ProfileName}.exe", $"-dataPath \"{selectedPack.Path.Trim('\\')}\"");
+            string basePath = Config.Get<string>("GamePath", "", ConfigScope.Game, ProfilesLibrary.ProfileName) + '\\';
+            string modDirName = "ModData\\" + selectedPack.Name;
+            string modDataPath = getModDataPath() + $"\\{selectedPack.Name}\\";
+            string additionalArgs = Config.Get<string>("CommandLineArgs", "", ConfigScope.Game);
+            
+            string steamAppIdPath = $"{basePath}steam_appid.txt";
+            if (System.IO.File.Exists(steamAppIdPath))
+            {
+                SaveFileDialog dialog = new SaveFileDialog();
```
Comment (wannkunstbeikor) [inline FrostyModManager/Windows/ManageModDataWindow.xaml.cs]: use FrostySaveFileDialog

Comment (wannkunstbeikor) [reply FrostyModManager/Windows/ManageModDataWindow.xaml.cs]: look at one of the plugins that has it on how to use

File: FrostyModManager/FrostyModManager.csproj (hunk 1 above)
Comment (shoushou1106) [reply FrostyModManager/FrostyModManager.csproj:305]: icon for save to shortcut button
that icon doesn't exist in modmanager

**Pull request merged**